src/main/java/bernie/commands/CommandHandler.java

```diff
@@ -0,0 +1,96 @@
+package bernie.commands;
+
+import java.util.Scanner;
+
+import bernie.enums.Type;
+import bernie.exceptions.BernieException;
+import bernie.parser.Parser;
+import bernie.storage.Storage;
+import bernie.tasks.TaskList;
+import bernie.ui.UiHandler;
+
+/**
+ * CommandHandler determines which Command to execute given the user input. Parser helps to
+ * parse the user input to help CommandHandler determine the respective Command.
+ */
+public class CommandHandler {
+    private TaskList tasks;
+    private UiHandler uiHandler;
+    private Storage storage;
+    private Parser parser;
+
+    /**
+     * Constructs the commandHandler to handle user inputs
+     */
+    public CommandHandler() {
+        this.tasks = new TaskList();
+        this.uiHandler = new UiHandler();
+        this.storage = new Storage();
+        this.parser = new Parser(tasks);
+    }
+
+    /**
+     * Commands are performed according to the input, done by creating the respective commands and
+     * executing it. Parser helps to determine the Type given the user input. uiHandler displays to
+     * the user the relevant message according to their input.
+     * Exceptions are caught and printed out for the user.
+     * @param input String, user input
+     * @return boolean value, indicating if the program will end or not.
+     */
+    public boolean handle (String input) {
```

Comment:
I like the functionality of this method, but as per Java coding standard, you could consider changing the method name or changing the return type. Also, a minor typo in which there shouldn't be a space between `handle` and `(String input)`.